web/client/codechecker_client/cmd/store.py

```diff
@@ -509,10 +514,13 @@ def assemble_zip(inputs, zip_file, client):
                 except KeyError:
                     zipf.write(f, file_path)
 
-        if collected_file_paths:
+        if necessary_blame_hashes:
+            file_paths = list(f for f, h in file_to_hash.items()
+                              if h in necessary_blame_hashes)
+
             LOG.info("Collecting blame information for source files...")
             try:
-                if assemble_blame_info(zipf, collected_file_paths):
+                if assemble_blame_info(zipf, file_paths):
```

Comment:
Can you please add a test case to see whether this functionality works.
You can create a temporary directory with a single source file, analyze and store it, after initiliazie this directory as a git repo, analyze and store it again and see whether blame informations are available.